src/bun.js/test/expect.zig

```diff
@@ -0,0 +1,3151 @@
+    scope: *DescribeScope,
+    not: bool = false,
+    resolves: ?Resolves = null,
+
+    pub const Resolves = enum(u1) {
+        resolves,
+        rejects,
+    };
+
+    pub fn getSignature(comptime matcher_name: string, comptime args: string, comptime not: bool) string {
+        const received = "<d>expect(<r><red>received<r><d>).<r>";
+        comptime if (not) {
+            return received ++ "not<d>.<r>" ++ matcher_name ++ "<d>(<r>" ++ args ++ "<d>)<r>";
+        };
+        return received ++ matcher_name ++ "<d>(<r>" ++ args ++ "<d>)<r>";
+    }
+
+    pub fn getFullSignature(comptime matcher: string, comptime args: string, comptime not: bool, comptime resolves: ?Resolves) string {
+        const fmt = "<d>expect(<r><red>received<r><d>).<r>" ++ if (resolves) |res|
+            switch (res) {
+                .resolves => if (not) "resolves<d>.<r>not<d>.<r>" else "resolves<d>.<r>",
+                .rejects => if (not) "rejects<d>.<r>not<d>.<r>" else "rejects<d>.<r>",
+            }
+        else if (not) "not<d>.<r>" else "";
+        return fmt ++ matcher ++ "<d>(<r>" ++ args ++ "<d>)<r>";
+    }
+
+    pub fn getNot(this: *Expect, thisValue: JSValue, _: *JSGlobalObject) callconv(.C) JSValue {
+        this.not = !this.not;
+        return thisValue;
+    }
+
+    pub fn getResolves(this: *Expect, thisValue: JSValue, globalThis: *JSGlobalObject) callconv(.C) JSValue {
+        if (this.resolves) |resolves| {
+            switch (resolves) {
+                .rejects => {
+                    globalThis.throw("Cannot chain .resolves() after .rejects()", .{});
+                    return .zero;
+                },
+                .resolves => {},
+            }
+        } else {
+            this.resolves = .resolves;
+        }
+        return thisValue;
+    }
+
+    pub fn getRejects(this: *Expect, thisValue: JSValue, globalThis: *JSGlobalObject) callconv(.C) JSValue {
+        if (this.resolves) |resolves| {
+            switch (resolves) {
+                .resolves => {
+                    globalThis.throw("Cannot chain .rejects() after .resolves()", .{});
+                    return .zero;
+                },
+                .rejects => {},
+            }
+        } else {
+            this.resolves = .rejects;
+        }
+        return thisValue;
+    }
+
+    pub fn getValue(this: *Expect, globalThis: *JSGlobalObject, thisValue: JSValue, comptime matcher_name: string, comptime matcher_args: string) ?JSValue {
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalThis.throw("{s}() must be called in a test", .{matcher_name});
+            return null;
+        }
+
+        const value = Expect.capturedValueGetCached(thisValue) orelse {
+            globalThis.throw("Internal error: the expect(value) was garbage collected but it should not have been!", .{});
+            return null;
+        };
+        value.ensureStillAlive();
+
+        if (this.resolves) |resolves| {
```

Comment:
The main question here is if `value` will need to be wrapped in a Promise as well. 

The pattern is 

```js
await expect(fetch).resolves.toBeInstanceOf(Response)
```


Sometimes, people do this:
```js
try {
   expect(123).toBe(456);
} catch (e) {

} 
```

So if they do that with a Promise and call `.then`, `.catch`, `.finally` etc, their code will throw an error.